### ISSUE TITLE
False alarm for new `async` properties and `implicit_getter`

### ISSUE THREAD

**Issue by basememara** [realm/SwiftLint#3684]
This triggers a violation for `implicit_getter`:

```swift
class DatabaseEntity {
    var isSynced: Bool {
        get async {
            await database.isEntitySynced(self)
        }
    }
}
```

There isn't another way to define an `async` property in Swift 5.5 so the getter is required here. I think if `get async` is declared for a property, it should not trigger a `implicit_getter` failure.

**Comment by letatas** [comment]
The problem is the same with `throws` and also with subscripts.

```swift
struct Test {
    subscript (value: Int) -> Int {
        get throws {
            if value == 0 {
                throw NSError()
            }
            else {
                return value
            }
        }
    }
}
```
